rpcs3/CMakeLists.txt

```diff
@@ -313,11 +313,7 @@ if(NOT LLVM_FOUND)
 else()
 	add_definitions(${LLVM_DEFINITIONS})
 	add_definitions(-DLLVM_AVAILABLE)
-	if(CMAKE_BUILD_TYPE STREQUAL "Release")
-		llvm_map_components_to_libnames(LLVM_LIBS core support executionengine object runtimedyld x86desc x86info scalaropts mcjit vectorize ipo x86codegen x86disassembler)
-	else()
-		llvm_map_components_to_libnames(LLVM_LIBS core support executionengine object runtimedyld x86desc x86info scalaropts mcjit vectorize ipo x86codegen x86disassembler mcdisassembler)
-	endif()
+	set(LLVM_LIBS LLVMMCJIT LLVMX86CodeGen)
```

Comment:
`set` is archaic and could break. Why not continue using `llvm_map_components_to_libnames` ? At least that's the suggested usage.

Reply:
How can it break?
The targets of the libraries are exactly what I wrote. They're not going to change.

Reply:
@Orphis If they change the name (depending on the build config) or something. Better use the provided function without explicit reason not to do it.

Reply:
It's the name of the target, not the name of the build artifact. Target names are stable.

Reply:
Again, it might work but why change from officially supported method?

Reply:
That's only a method that's used to map llvm-config components to CMake targets.
Nowhere does it say you can't use the CMake targets directly. Lots of projects are integrating with the LLVM codebase and they have to use those targets.
Renaming those for no reason would break them and they don't want to do that, so it's a part of the API now.